src/Eloquent/HasDocumentVersion.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Eloquent;
+
+trait HasDocumentVersion
+{
+    /**
+     * Auto call on model instance as booting
+     *
+     * @return void
+     */
+    public static function bootHasDocumentVersion(): void
+    {
+        static::saving(function ($model) {
+          $version = $model->getDocumentVersion();
+
+          if (!$version) {
+              $model->setDocumentVersion(static::getCurrentVersion());
+          }
+        });
+
+        static::retrieved(function ($model) {
+            $model->migrateDocumentVersion($model->getDocumentVersion());
+        });
+    }
+
+    /**
+     * migrate model document version schema
+     *
+     * @param int|null $fromVersion
+     * @return void
+     */
+    public function migrateDocumentVersion($fromVersion): void
+    {
+    }
```

Comment:
When the phpdoc duplicate the native types, you can omit them, same for other methods.

Also updating the comment.
```suggestion
    /**
     * Migrate document to the current schema version.
     */
    public function migrateDocumentVersion(int $fromVersion): void
    {
    }
```

Reply:
Understood